src/main/java/net/fabricmc/loom/api/RunConfiguration.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * This file is part of fabric-loom, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2025 FabricMC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.loom.api;
+
+import java.util.Map;
+
+import org.gradle.api.Named;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.SourceSet;
+import org.jetbrains.annotations.ApiStatus;
+
+import net.fabricmc.loom.util.Constants;
+import net.fabricmc.loom.util.Platform;
+
+/**
+ * Represents a run configuration for Minecraft, these can presented via an IDE run configuration or a Gradle task.
+ *
+ * <p>This is used to configure how Minecraft is run, including JVM arguments, program arguments, environment variables,
+ * and the main class to run.
+ */
+public interface RunConfiguration extends Named {
+	/**
+	 * The full name of the run configuration, i.e. 'Minecraft Client'.
+	 *
+	 * <p>By default this is determined from the base name.
+	 *
+	 * <p>Note: unless the project is the root project (or {@link #getAppendProjectPathToConfigName()} is disabled),
+	 * the project path will be appended automatically, e.g. 'Minecraft Client (:some:project)'.
+	 */
+	Property<String> getConfigurationName();
```

Comment:
`getDisplayName` might be clearer (`RunConfiguration.getName()` from `Named` is already a "configuration name")

Reply:
Or getIdeName?